mmv1/products/compute/terraform.yaml

```diff
@@ -2274,6 +2274,7 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         skip_vcr: true
     custom_code: !ruby/object:Provider::Terraform::CustomCode
       update_encoder: templates/terraform/update_encoder/reservation.go.erb
+      pre_update: templates/terraform/pre_update/shared_reservation_update.go.erb
```

Comment:
`pre_update` methods aren't supported by fields w/ custom update methods (as if there are multiple updatable fields, it's unclear whether it should be run per-field or once globally)

We'll need to unmark the resource as `input: true`, and mark any fields that don't support update as `input: true`. That will signal that the resource is updatable but the majority of fields aren't. Note that `resize` is still a custom update method- that can be left alone, I think mixed updates work (but it has been a while)